echopop/spatial/transect.py

```diff
@@ -0,0 +1,332 @@
+    and replaces/imputes with alternative lengths/areas.
+    """
+
+    # Create dataframe copy
+    transect_data_copy = transect_data.copy()
+
+    # Calculate the along-transect interval distance
+    # ---- Entire dataframe
+    transect_data_copy["interval"] = transect_data_copy["vessel_log_start"].diff(periods=-1).abs()
+    # ---- Replace the trailing NaN
+    transect_data_copy["interval"] = transect_data_copy["interval"].replace(
+        np.nan,
+        transect_data_copy["vessel_log_end"].iloc[-1]
+        - transect_data_copy["vessel_log_start"].iloc[-1],
+    )
+
+    # Replace (likely) erroneous interval lengths associated at the ends of each transect
+    # ---- Calculate median interval
+    median_interval = np.median(transect_data_copy["interval"])
+    # ---- Find indices where interval deviations from the median exceed the difference threshold
+    transect_data_copy["interval"] = np.where(
+        np.abs(transect_data_copy["interval"] - median_interval) > interval_threshold,
+        transect_data_copy["vessel_log_end"] - transect_data_copy["vessel_log_start"],
+        transect_data_copy["interval"],
+    )
+
+    # Calculate the interval area
+    transect_data_copy["interval_area"] = (
+        transect_data_copy["interval"] * transect_data_copy["transect_spacing"]
+    )
+
+    # Filter out unnecessary columns and return output
+    return transect_data_copy.filter(
+        regex="^(?=transect_num|latitude|longitude|stratum_|haul_|interval_area|NASC_).*"
+    )
+
+
+def save_transect_coordinates(transect_data: pd.DataFrame):
+
+    # Extract transect numbers, coordinates, and strata
+    return transect_data[
+        [
+            "transect_num",
+            "stratum_num",
+            "stratum_inpfc",
+            "longitude",
+            "latitude",
+            "transect_spacing",
+        ]
+    ]
+
+
+def edit_transect_columns(transect_dict: dict, settings_dict: dict):
+
+    # Define the current stratum definition used for the transect data
+    new_stratum = settings_dict["stratum_name"]
+
+    # Prepare the transect data to be updated
+    # ---- Call transect data
+    transect_data = transect_dict["acoustics"]["adult_transect_df"].copy()
+    # -------- Set index
+    transect_data.set_index(["transect_num", "longitude", "latitude"], inplace=True)
+    # ---- Call full coordinate information
+    transect_info = transect_dict["coordinates"].copy()
+    # -------- Set index
+    transect_info.set_index(["transect_num", "longitude", "latitude"], inplace=True)
+
+    # Adjust stratum values
+    # ---- Change strata values
+    transect_info = transect_info[[new_stratum, "transect_spacing"]]
+    # ---- Select analysis variable
+    transect_info[f"{settings_dict['variable']}"] = transect_data[f"{settings_dict['variable']}"]
+
+    # Return the output
+    return transect_info.reset_index()
+
+
+def transect_distance(transect_data: pd.DataFrame):
```

Comment:
I kinda prefer `transect_spatial` or `transect_spatial_features` over `transect_distance` for the function name